mlflow/store/artifact/unity_catalog_oss_models_artifact_repo.py

```diff
@@ -0,0 +1,159 @@
+                "models in the Unity Catalog. We recommend that you access the Unity Catalog "
+                "from the current Databricks workspace instead."
+            )
+        self.registry_uri = registry_uri
+        self.client = MlflowClient(registry_uri=self.registry_uri)
+        try:
+            spark = _get_active_spark_session()
+        except Exception:
+            pass
+        model_name, self.model_version = get_model_name_and_version(self.client, artifact_uri)
+        self.model_name = get_full_name_from_sc(model_name, spark)
+
+    def _get_blob_storage_path(self):
+        return self.client.get_model_version_download_uri(self.model_name, self.model_version)
+
+    def _get_scoped_token(self, lineage_header_info=None):
+        extra_headers = {}
+        if lineage_header_info:
+            header_json = message_to_json(lineage_header_info)
+            header_base64 = base64.b64encode(header_json.encode())
+            extra_headers[_DATABRICKS_LINEAGE_ID_HEADER] = header_base64
+
+        oss_creds = get_oss_host_creds(
+            self.registry_uri
+        )  # TODO: Discuss & Implement OSS Host Creds properly
+        oss_endpoint, oss_method = _METHOD_TO_INFO_OSS[GenerateTemporaryModelVersionCredentialsOSS]
+        [catalog_name, schema_name, model_name] = self.model_name.split(
+            "."
+        )  # self.model_name is actually the full name
+        oss_req_body = message_to_json(
+            GenerateTemporaryModelVersionCredentialsOSS(
+                catalog_name=catalog_name,
+                schema_name=schema_name,
+                model_name=model_name,
+                version=int(self.model_version),
+                operation=MODEL_VERSION_OPERATION_READ_OSS,
+            )
+        )
+        oss_response_proto = TemporaryCredentials()
+        return call_endpoint(
+            host_creds=oss_creds,
+            endpoint=oss_endpoint,
+            method=oss_method,
+            json_body=oss_req_body,
+            response_proto=oss_response_proto,
+            extra_headers=extra_headers,
+        )
+
+    def _get_artifact_repo(self, lineage_header_info=None):
+        """
+        Get underlying ArtifactRepository instance for model version blob
+        storage
+        """
+        scoped_token = self._get_scoped_token(lineage_header_info=lineage_header_info)
+        blob_storage_path = self._get_blob_storage_path()
+        return get_artifact_repo_from_storage_info(
+            storage_location=blob_storage_path,
+            scoped_token=scoped_token,
+            base_credential_refresh_def=self._get_scoped_token,
+            is_oss=True,
+        )
+
+    def list_artifacts(self, path=None):
+        return self._get_artifact_repo().list_artifacts(path=path)
+
+    def download_artifacts(self, artifact_path, dst_path=None, lineage_header_info=None):
+        return self._get_artifact_repo(lineage_header_info=lineage_header_info).download_artifacts(
```

Comment:
We probably need to pass in the artifact path to `_get_artifact_repo` here so that inside that method, we can flag not getting a scoped token and leaving the scoped token empty when we call into `get_artifact_repo_from_storage_info` we can have that method return a LocalArtifactRepository.

Reply:
making the fix rn